cmd/vcluster/cmd/start.go

```diff
@@ -302,9 +303,20 @@ func ExecuteStart(options *context2.VirtualClusterOptions) error {
 
 	if ctx.Options.LeaderElect {
 		err = leaderelection.StartLeaderElection(ctx, scheme, func() error {
+			// check apiservice for metrics server
+			err := metricsapiservice.RegisterOrDeregisterAPIService(ctx.Context, ctx.Options, virtualClusterConfig)
+			if err != nil {
+				klog.Errorf("Error registering metrics apiservice: %v", err)
+			}
 			return startControllers(ctx, &rawConfig, serverVersion)
 		})
 	} else {
+		// check apiservice for metrics server
+		err = metricsapiservice.RegisterOrDeregisterAPIService(ctx.Context, ctx.Options, virtualClusterConfig)
+		if err != nil {
+			klog.Errorf("Error registering metrics apiservice: %v", err)
+		}
+
```

Comment:
Would it cause any problems if we called `metricsapiservice.RegisterOrDeregisterAPIService` from a goroutine instead of blocking and let `startControllers` be called in parallel?

Reply:
I don't think it should cause any issue, moved it to a goroutine